kfs-cam/src/main/java/edu/arizona/kfs/module/cam/document/AssetMaintainableImpl.java

```diff
@@ -0,0 +1,111 @@
+package edu.arizona.kfs.module.cam.document;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.kns.document.MaintenanceDocument;
+import org.kuali.rice.krad.bo.Note;
+import org.kuali.rice.krad.service.KRADServiceLocator;
+import org.kuali.rice.krad.service.NoteService;
+import org.kuali.rice.krad.util.GlobalVariables;
+
+import edu.arizona.kfs.sys.KFSConstants;
+
+@SuppressWarnings("deprecation")
+public class AssetMaintainableImpl extends org.kuali.kfs.module.cam.document.AssetMaintainableImpl {
+
+    private static final long serialVersionUID = 5710219294064590806L;
+
+    @Override
+    public void processAfterEdit(MaintenanceDocument document, Map<String, String[]> parameters) {
+        List<Note> notes = new ArrayList<Note>();
+        if (getBusinessObject().getObjectId() != null) {
+            NoteService noteService = KRADServiceLocator.getNoteService();
+            notes = noteService.getByRemoteObjectId(getBusinessObject().getObjectId());
+        }
+
+        setAssetCreateAndUpdateNote(notes, KFSConstants.CreateAndUpdateNotePrefixes.CHANGE);
+        document.setNotes(notes);
+
+        super.processAfterEdit(document, parameters);
+    }
+
+    @Override
+    public void setGenerateDefaultValues(String docTypeName) {
+        super.setGenerateDefaultValues(docTypeName);
+
+        List<Note> notes = new ArrayList<Note>();
```

Comment:
This generates a "redundant initialization" warning in my IDE, but I realize it might not in yours. Though, even just looking at this snippet set off a red flag for me. This was due to a list being instantiated on 39, yet it is then almost immediately set unconditionally at line 43 (so long as this object has been saved before). A quick fix would be to declare the left-hand side of line 39 to the right hand side on line 43, and finally just delete line 39. That would avoid redundant double assignment, but more importantly, communicates that this list really isn't used outside of the if statement on line 45. The list is never set directly on an object, the list is never sent to the DB, and the list is not returned as a result.

Also while we're close in code, I regard the method "getNewBoNoteForAdding(...)" on line 46 to be sub-optimal, since its name neither communicates what all it's actually doing (generate + persist new note),  nor does the method limit itself to only one discrete task (the latter admittedly can sometimes get subjective). Was this part of ported code, or was this written by a UA/EC dev in-house?